src/dag.ts

```diff
@@ -68,25 +71,27 @@ export function iterate<T>(nodes: INode<T>[], edges: IEdge[],
     return unvisited === false ? childNodes : childNodes.filter(filterNotVisited);
   }
 
-  let i = 0;
-  nodeStack.push(getFirstNode());
+  Array.prototype.push.apply(nodeStack, getFirstNodes());
```

Comment:
would be great to add a quick comment here:
`// push all starting nodes on the stack`